### PR TITLE
[TextFields] Floating placeholder font changes after users starts typing.

### LINKED ISSUE #4189: [Textfield] Floating Placeholder text changes size when you type in the 2nd word into the text field
In a controller make a MDCTextfield a wrap a MDCTextInputControllerFilled controller on it

let textField:BITextField = BITextField()
MDCTextInputControllerFilled(textInput: textField)

In the text field type 1 letter in, the floating placeholder text will get smaller and go on top

When you type in a 2nd letter in, the floating placeholder text will get smaller again.

Expected behavior: The floating placeholder text should remain the same size and not change

---
#### Internal data
- Associated internal bug: [b/117178829](http://b/117178829)

### PR DESCRIPTION
**Pull request by mohammadcazig**
Fixing the order of operations for textfields update. Setting TextInput Font causes the placeholder font to be updated to the same font. so we update placeholderFont after updating the text input font to get the desired font for placeholder.

close #4189

### CODE REVIEW

**Review by jverkoey — APPROVED** [review]

**Comment by jverkoey** [comment]
Are there tests for this?

**Comment by willlarche** [comment]
Needs comments to prevent people undoing it by accident later.

**Review by willlarche — APPROVED** [review]

File: components/TextFields/src/MDCTextInputControllerFullWidth.m [hunk 1]
```diff
@@ -842,10 +842,10 @@ - (void)updateLayout {
     return;
   }
 
-  [self updatePlaceholder];
   [self updateLeadingUnderlineLabel];
   [self updateTrailingUnderlineLabel];
   [self updateTextInput];
+  [self updatePlaceholder];
```
Comment (willlarche) [inline components/TextFields/src/MDCTextInputControllerFullWidth.m]: Missing comment.

**Review by willlarche — DISMISSED** [review]

File: components/TextFields/src/MDCTextInputControllerFullWidth.m (hunk 1 above)
Comment (mohammadcazig) [reply components/TextFields/src/MDCTextInputControllerFullWidth.m]: Done.

File: components/TextFields/tests/unit/TextInputControllerBaseTests.m [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ limitations under the License.
+ */
+
+#import <XCTest/XCTest.h>
+
+#import "MaterialTextFields.h"
+
+@interface MDCTextInputControllerBase ()
+
+- (void)updateLayout;
```
Comment (willlarche) [inline components/TextFields/tests/unit/TextInputControllerBaseTests.m:23]: This just blew my mind.

**Pull request merged**